### PR TITLE
Added RetroDECK Manifest

### PR DESCRIPTION
**Pull request by XargonWan**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *I am using the permissions of the flatpak where I took the submodules from, all of them already available on flathub*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Review by hfiguiere — CHANGES_REQUESTED** [review]
this is never build. You need an actual manifest.

**Comment by XargonWan** [comment]
> this will never build. You need an actual manifest.

Omg sorry, updated.

File: com.xargon.retrodeck.yml [hunk 1]
```diff
@@ -0,0 +1,717 @@
+app-id: com.xargon.retrodeck
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:1]: the domain `xargon.com` seems to just be domain squatted. I don't think this fit the criteria for app-id.

File: com.xargon.retrodeck.yml [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+   - --filesystem=/run/media
+   - --allow=multiarch
+   - --talk-name=org.freedesktop.ScreenSaver
+   - --talk-name=org.freedesktop.PowerManagement.Inhibit
+   - --talk-name=org.freedesktop.login1
+   #- --filesystem=host
+   - --filesystem=xdg-run/app/com.discordapp.Discord:create
+  # Yuzu
+   - --filesystem=home:ro
+   - --filesystem=/run/media:ro
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:31]: these are redundant with the others.

File: com.xargon.retrodeck.yml [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+
+cleanup:
+  # ES-DE
+  - /include
+  - /share/ffmpeg
+  - /lib/cmake
+  - /lib/debug
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:38]:
```suggestion
```

File: com.xargon.retrodeck.yml [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+      - --with-libxml-libs-prefix=/usr/lib
+      - --with-libxml-include-prefix=/usr/include/libxml2
+    sources:
+      - type: archive
+        url: https://downloads.sourceforge.net/xmlstar/xmlstarlet-1.6.1.tar.gz
+        sha256: 15d838c4f3375332fd95554619179b69e4ec91418a3a5296e7c631b7ed19e7ca
+      - type: shell
+        commands:
+          - cp -p /usr/share/automake-*/config.{sub,guess} .;
+          - autoreconf -vfi;
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:72]:
```suggestion
          - cp -p /usr/share/automake-*/config.{sub,guess} .
          - autoreconf -vfi
```

File: com.xargon.retrodeck.yml [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+    post-install:
+      - ln -s "xml" "${FLATPAK_DEST}/bin/xmlstarlet" ||:;
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:74]:
```suggestion
      - ln -s "xml" "${FLATPAK_DEST}/bin/xmlstarlet" ||:
```

File: com.xargon.retrodeck.yml [hunk 6]
```diff
@@ -0,0 +1,717 @@
...
+    cleanup: ["*"]
+
+  - name: kdialog
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:78]:
```suggestion
    buildsystem: cmake-ninja
```

File: com.xargon.retrodeck.yml [hunk 7]
```diff
@@ -0,0 +1,717 @@
...
+    sources:
+    - type: git
+      url: https://github.com/KDE/kdialog.git
+      tag: v22.03.80
+
+  # ES-DE
+
+  - name: fdk-aac
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:86]: this is in the runtime

File: com.xargon.retrodeck.yml [hunk 8]
```diff
@@ -0,0 +1,717 @@
...
+    buildsystem: cmake
+    sources:
+    - type: git
+      url: https://git.code.sf.net/p/opencore-amr/fdk-aac.git
+      tag: v2.0.2
+
+  - name: libvpx
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:93]: isn't 1.10 (it's in the runtime) enough?

File: com.xargon.retrodeck.yml [hunk 9]
```diff
@@ -0,0 +1,717 @@
...
+    - --disable-examples
+    - --disable-docs
+    - --enable-pic
+    - --enable-vp9-highbitdepth
+    sources:
+    - type: git
+      url: https://github.com/webmproject/libvpx.git
+      tag: v1.11.0
+
+  - name: ogg
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:104]: this is in the runtime

File: com.xargon.retrodeck.yml [hunk 10]
```diff
@@ -0,0 +1,717 @@
...
+    buildsystem: cmake
+    config-opts:
+    - -DCMAKE_POSITION_INDEPENDENT_CODE=on
+    - -DINSTALL_DOCS=off
+    sources:
+    - type: git
+      url: https://github.com/xiph/ogg.git
+      tag: v1.3.5
+
+  - name: vorbis
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:114]: this is in the runtime

File: com.xargon.retrodeck.yml [hunk 11]
```diff
@@ -0,0 +1,717 @@
...
+    buildsystem: cmake
+    config-opts:
+    - -DBUILD_SHARED_LIBS=on
+    sources:
+    - type: git
+      url: https://gitlab.xiph.org/xiph/vorbis.git
+      tag: v1.3.7
+
+  - name: opus
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:123]: this is in the runtime

File: com.xargon.retrodeck.yml [hunk 12]
```diff
@@ -0,0 +1,717 @@
...
+    - -DBUILD_SHARED_LIBS=on .
+    sources:
+    - type: archive
+      url: https://github.com/zeux/pugixml/releases/download/v1.11.4/pugixml-1.11.4.tar.gz
+      sha256: 8ddf57b65fb860416979a3f0640c2ad45ddddbbafa82508ef0a0af3ce7061716
+
+  - name: emulationstation-de
+    buildsystem: cmake
+    config-opts:
+      - -DCMAKE_INSTALL_PREFIX=/app
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:177]:
```suggestion
```

this is implicit

File: com.xargon.retrodeck.yml [hunk 13]
```diff
@@ -0,0 +1,717 @@
...
+    - type: archive
+      url: http://downloads.sourceforge.net/freeimage/FreeImage3180.zip
+      sha256: f41379682f9ada94ea7b34fe86bf9ee00935a3147be41b6569c9605a53e438fd
+    - type: shell
+      commands:
+      - sed -i 's|-o root -g root ||' ./Makefile.gnu
+      - sed -i 's|/usr|/app|' ./Makefile.gnu
+
+  - name: pugixml
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:166]:
```suggestion
    buildsystem: cmake-ninja
```

File: com.xargon.retrodeck.yml [hunk 14]
```diff
@@ -0,0 +1,717 @@
...
+    config-opts:
+    - -DBUILD_SHARED_LIBS=on .
+    sources:
+    - type: archive
+      url: https://github.com/zeux/pugixml/releases/download/v1.11.4/pugixml-1.11.4.tar.gz
+      sha256: 8ddf57b65fb860416979a3f0640c2ad45ddddbbafa82508ef0a0af3ce7061716
+
+  - name: emulationstation-de
+    buildsystem: cmake
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:175]:
```suggestion
    buildsystem: cmake-ninja
```

File: com.xargon.retrodeck.yml [hunk 15]
```diff
@@ -0,0 +1,717 @@
...
+
+  - name: retroarch
+    config-opts:
+      - '--enable-dbus'
+    make-args:
+      - GLOBAL_CONFIG_DIR=${FLATPAK_DEST}/etc
+    sources:
+      - type: git
+        url: https://github.com/XargonWan/RetroDECK.git
+        branch: 9d8b5ad985c5b29980810fdf05e1a534d2070ce2
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:211]:
```suggestion
        commit: 9d8b5ad985c5b29980810fdf05e1a534d2070ce2
```

File: com.xargon.retrodeck.yml [hunk 16]
```diff
@@ -0,0 +1,717 @@
...
+
+  - name: art-book-next
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/emulationstation/themes/art-book-next-de/
+      - mv -f * ${FLATPAK_DEST}/share/emulationstation/themes/art-book-next-de/
+    sources:
+      - type: git
+        url: https://github.com/anthonycaccese/art-book-next-de.git
+        branch: 4452f8356819d3cd079549b113fb15edb919a0df
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:196]:
```suggestion
        commit: 4452f8356819d3cd079549b113fb15edb919a0df
```

File: com.xargon.retrodeck.yml [hunk 17]
```diff
@@ -0,0 +1,717 @@
...
+    buildsystem: cmake
+    config-opts:
+      - -DCMAKE_INSTALL_PREFIX=/app
+    cleanup:
+      - es-app
+      - es-core
+    sources:
+    - type: git
+      url: https://gitlab.com/leonstyhre/emulationstation-de.git
+      branch: stable-1.2.3
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:184]: you need an explicit tag or commit. Not a branch.

File: com.xargon.retrodeck.yml [hunk 18]
```diff
@@ -0,0 +1,717 @@
...
+      - '--enable-dbus'
+    make-args:
+      - GLOBAL_CONFIG_DIR=${FLATPAK_DEST}/etc
+    sources:
+      - type: git
+        url: https://github.com/XargonWan/RetroDECK.git
+        branch: 9d8b5ad985c5b29980810fdf05e1a534d2070ce2
+      - type: git
+        url: https://github.com/libretro/RetroArch.git
+        branch: de4e56ecf28a09057734f8624af6d62f3088734f
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:214]:
```suggestion
        commit: de4e56ecf28a09057734f8624af6d62f3088734f
```

File: com.xargon.retrodeck.yml [hunk 19]
```diff
@@ -0,0 +1,717 @@
...
+      - type: file
+        path: rd-submodules/retroarch/org.libretro.RetroArch.appdata.xml
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:216]: `rd-submodules` doesn't exist in the repository

File: com.xargon.retrodeck.yml [hunk 20]
```diff
@@ -0,0 +1,717 @@
...
+        mv ${FLATPAK_DEST}/share/pixmaps/retroarch.svg
+        ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/
+      - rmdir --ignore-fail-on-non-empty ${FLATPAK_DEST}/share/pixmaps/
+      - mkdir -p ${FLATPAK_DEST}/etc
+      - >-
+        sed s:@prefix@:${FLATPAK_DEST}:g retroarch.cfg >
+        ${FLATPAK_DEST}/etc/retroarch.cfg
+      - mkdir -p ${FLATPAK_DEST}/share/appdata
+      - cp org.libretro.RetroArch.appdata.xml ${FLATPAK_DEST}/share/appdata
+    modules:
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:231]: none of these manifests are in the repository

File: com.xargon.retrodeck.yml [hunk 21]
```diff
@@ -0,0 +1,717 @@
...
+      - libpng/libpng-1.6.35.json
+      - nvidia-cg-toolkit/nvidia-cg-toolkit-3.1.0013.json
+      - shared-modules/SDL/SDL-1.2.15.json
+      - shared-modules/SDL/SDL_image-1.2.12.json
+      - shared-modules/SDL/SDL_mixer-1.2.12.json
+      - shared-modules/SDL/SDL_net-1.2.8.json
+      - shared-modules/SDL/SDL_ttf-2.0.11.json
+      - shared-modules/libusb/libusb.json
+      - shared-modules/gudev/gudev.json
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:240]: you need to add the shared-modules repository as a git submodule.

File: com.xargon.retrodeck.yml [hunk 22]
```diff
@@ -0,0 +1,717 @@
...
+  # RetroArch - END
+
+  # Not part of the offical RetroArch manifest
+  - name: retroarch-cores
+    buildsystem: simple
+    build-commands:
+        - mkdir -p /app/share/libretro/cores/
+        - mv ./RetroArch-Linux-x86_64.AppImage.home/.config/retroarch/cores/* /app/share/libretro/cores/
+    cleanup:
+        - RetroArch_cores.7z
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:328]: cleanup does do what you think it does
```suggestion
```

File: com.xargon.retrodeck.yml [hunk 23]
```diff
@@ -0,0 +1,717 @@
...
+      - >-
+        mv /app/share/mime/packages/yuzu.xml
+        /app/share/mime/packages/org.yuzu_emu.yuzu.xml
+      - >-
+        sed 's/yuzu/org.yuzu_emu.yuzu/g' -i
+        /app/share/mime/packages/org.yuzu_emu.yuzu.xml
+    sources:
+      - type: git
+        url: https://github.com/XargonWan/RetroDECK.git
+        branch: 9d8b5ad985c5b29980810fdf05e1a534d2070ce2
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:489]:
```suggestion
        commit: 9d8b5ad985c5b29980810fdf05e1a534d2070ce2
```

File: com.xargon.retrodeck.yml [hunk 24]
```diff
@@ -0,0 +1,717 @@
...
+      - /share/doc
+    post-install:
+      - >-
+        desktop-file-edit --set-key=Exec --set-value=pcsx2
+        /app/share/applications/PCSX2.desktop
+      - >-
+        install -Dm644 ../AppIcon128.png
+        /app/share/icons/hicolor/128x128/apps/PCSX2.png
+      - >-
+        install -Dm644 ../net.pcsx2.PCSX2.metainfo.xml
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:627]: same, there can only be one appstream file.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
as it is it won't build

**Comment by hfiguiere** [comment]
also, this package many app that are already on flathub.... what's the goal here?

File: com.xargon.retrodeck.yml (hunk 1 above)
Comment (XargonWan) [reply com.xargon.retrodeck.yml:1]: It`s not realted with that website, atm my only website is github, a new one is in the making

File: shared-modules [hunk 25]
```diff
@@ -0,0 +1 @@
+/home/deck/RetroDECK/shared-modules
```
Comment (hfiguiere) [inline shared-modules:1]: not sure what you are doing here. This is supposed to be a gitsubmodule.

File: desktop_entry [hunk 26]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Name=RetroDECK
+GenericName=Emulation bundle for Steam Deck
+Type=Application
+Comment=All you need for emulation on Steam Deck
+Icon=com.xargon.retrodeck
+Exec=
```
Comment (hfiguiere) [inline desktop_entry:7]: you are missing an executable here.

File: desktop_entry [hunk 27]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline desktop_entry:1]: why is this file not properly named?

File: com.xargon.retrodeck.yml [hunk 28]
```diff
@@ -0,0 +1,697 @@
...
+        - cp tools-gamelist.xml ${FLATPAK_DEST}/retrodeck/
+        
+        - cp retrodeck.sh /app/bin/retrodeck.sh
+        - chmod +x /app/bin/retrodeck.sh
+
+        # Desktop entry
+        - cp desktop_entry /app/share/applications/com.xargon.retrodeck.desktop
+
+    #cleanup: ['*']
+    sources:
```
Comment (hfiguiere) [inline com.xargon.retrodeck.yml:679]: you don't have a single source that is either git or archive....

**Review by hfiguiere — CHANGES_REQUESTED** [review]
many question still unanswered.

shared-modules still missing.

File: com.xargon.retrodeck.yml (hunk 1 above)
Comment (hfiguiere) [reply com.xargon.retrodeck.yml:1]: then why are you using it?

**Pull request closed without merging**